src/component.cpp

```diff
@@ -181,6 +183,62 @@ void Component::removeAllVariables()
     mPimpl->mVariables.clear();
 }
 
+*/
+
+bool Component::removeVariable(size_t index)
+{
+    if (index < mPimpl->mVariables.size()) {
+        VariablePtr v = mPimpl->mVariables.at(index);
+        ModelPtr model = v->parentModel();
+        v->clearParent();
+        v->setParent(model);
```

Comment:
OK, just had another look at this.  It was because the only way to remove the parent **component** was to call the clearParent function, which wipes **both** the component parent as well as the model parent.  I put these setParent lines there to make sure that the model parent wasn't deleted, as that wasn't the point of the function?

Reply:
(it doesn't mean that the model parent will be used, only that it's not deleted while deleting the parent component)

Reply:
This is part of the problem with this code (solved by my other PR shameless promotion) there should in theory only ever be one parent set so clearing the parent should only effect the one that is active if any.

Reply:
Either way... whatever is here should be consistent.  Agreed that it can be fixed in another branch, but there's no point doing it in two places, that will just get super confusing!